test/stats/stats_aggregate_test.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Tests for unique type declaration rules validator.
+
+#include <string>
+
+#include "source/spirv_stats.h"
+#include "test_fixture.h"
+#include "unit_spirv.h"
+
+namespace {
+
+using libspirv::SpirvStats;
+using spvtest::ScopedContext;
+
+// Calls libspirv::AggregateStats for binary compiled from |code|.
+void CompileAndAggregateStats(const std::string& code, SpirvStats* stats,
+                    spv_target_env env = SPV_ENV_UNIVERSAL_1_1) {
+  ScopedContext ctx(env);
+  spv_binary binary;
+  ASSERT_EQ(SPV_SUCCESS, spvTextToBinary(
+      ctx.context, code.c_str(), code.size(), &binary, nullptr));
+
+  ASSERT_EQ(SPV_SUCCESS, AggregateStats(*ctx.context, binary->code,
+                                        binary->wordCount, nullptr, stats));
+  spvBinaryDestroy(binary);
+}
+
+TEST(AggregateStats, CapabilityHistogram) {
+  const std::string code1 = R"(
+OpCapability Addresses
+OpCapability Kernel
+OpCapability GenericPointer
+OpCapability Linkage
+OpMemoryModel Physical32 OpenCL
+)";
+
+  const std::string code2 = R"(
+OpCapability Shader
+OpCapability Linkage
+OpMemoryModel Logical GLSL450
+)";
+
+  SpirvStats stats;
+
+  CompileAndAggregateStats(code1, &stats);
+  EXPECT_EQ(4, stats.capability_hist.size());
```

Comment:
Signed vs. unsigned comparison error in Clang.  Recommend you add "u" suffix to all these literal numbers that compare to counts.